### PR TITLE
Semantic zoom

### PR DESCRIPTION
**Pull request by mischkew**
works til fontsize 36.
hint from tobias does not work.

### CODE REVIEW

File: packages/Path-View-Core.package/SClassDiagram.class/instance/enlargeClassNamesTo..st [hunk 1]
```diff
@@ -0,0 +1,6 @@
+zooming
+enlargeClassNamesTo: aHeight
+	
+	"sm - this should build large font sizes, but does not work in my image..."
+	"(TextStyle named: SClassNameMorph fontName) addNewFontSize: aHeight."
```
Comment (01Track1mp3) [inline packages/Path-View-Core.package/SClassDiagram.class/instance/enlargeClassNamesTo..st]: this breaks: some initialize function not defined

File: packages/Path-View-Core.package/SClassDiagram.class/instance/enableSemanticZooming.st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+zooming
+enableSemanticZooming
+
+	self classMorphs do: [ :classMorph | classMorph showLabels: self labelsHidden not ].
```
Comment (nicoring) [inline packages/Path-View-Core.package/SClassDiagram.class/instance/enableSemanticZooming.st]: duplicate to:
```smalltalk
setClassMorphsHidden: aBoolean
	self classMorphs do: [ :classMorph | classMorph showLabels: aBoolean ] 
```

File: packages/Path-View-Core.package/SClassDiagram.class/instance/setSemanticZooming..st [hunk 3]
```diff
@@ -0,0 +1,11 @@
+zooming
+setSemanticZooming: aNumber
+
+	| fontSize |
+	self isSemanticZooming
```
Comment (nicoring) [inline packages/Path-View-Core.package/SClassDiagram.class/instance/setSemanticZooming..st]: can this happen? For me it looks like that this method is only called if semanticZooming is enabled

File: packages/Path-View-Core.package/SClassNameMorph.class/README.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+I am an ExpandableEntry morph, which is part of an actual class morph on the SClassDiagram. I show the name of the class represented by the SClassDiagramMorph.
+
```
Comment (nicoring) [inline packages/Path-View-Core.package/SClassNameMorph.class/README.md]: :+1:

File: packages/Path-View-Tests.package/SClassDiagramTest.class/instance/testDefaultZoomingWithClassMorphOnDiagram.st [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+	| font textStyle |
+	SScrollPane forMorph: instance.
+	self addClass: klass.
+
+	textStyle := TextStyle named: SClassNameMorph fontName.
+	font := textStyle fontOfPointSize: SClassNameMorph defaultFontSize.
+	
+	self 
+		assert: [ (self getActualFontSize) = font pointSize ]
```
Comment (nicoring) [inline packages/Path-View-Tests.package/SClassDiagramTest.class/instance/testDefaultZoomingWithClassMorphOnDiagram.st]: I think you can remove the braces.

File: packages/Path-View-Tests.package/SClassDiagramTest.class/instance/testZoomingWithClassMorphOnDiagram.st [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+	| zoomLevel |
+
+	zoomLevel := (klass zoomMax + klass zoomMin) / 2.
+
+	SScrollPane forMorph: instance.
+	self addClass: klass.
+	instance zoomTo: zoomLevel.
+	
+	Transcript show: (self getActualFontSize) asString, (instance classMorphs first nameMorph labelMorph contents) asString, (instance fontSizeFromScale: zoomLevel); cr.
```
Comment (nicoring) [inline packages/Path-View-Tests.package/SClassDiagramTest.class/instance/testZoomingWithClassMorphOnDiagram.st]: pls remove

File: packages/Path-View-Core.package/SClassDiagram.class/instance/setSemanticZooming..st (hunk 3 above)
Comment (mischkew) [inline packages/Path-View-Core.package/SClassDiagram.class/instance/setSemanticZooming..st]: you are absolutely right, but this check makes sure, i only traverse the diagram graph once to collapse all morphs. i renamed this to isAlreadySemanticZooming which should make things more clearly

File: packages/Path-View-Core.package/SClassDiagram.class/instance/enableSemanticZooming.st (hunk 2 above)
Comment (mischkew) [inline packages/Path-View-Core.package/SClassDiagram.class/instance/enableSemanticZooming.st]: good eyes! but this method has no senders, so i removed it

**Pull request merged**